### ISSUE TITLE
F2 Score and mAP metric for object detection model. 

### ISSUE THREAD

**Issue by pure-rgb** [keras-team/keras#15811]
I didn't find a metric regarding the object detection task in keras. 

How to implement the above two metrics in keras and use them? In this competition, https://www.kaggle.com/c/tensorflow-great-barrier-reef/overview/evaluation it uses F2 score for its object detection task but I don't know how to evaluate my model locally with this score? 

> The metric sweeps over IoU thresholds in the range of 0.3 to 0.8 with a step size of 0.05, calculating an F2 score at each threshold. For example, at a threshold of 0.5, a predicted object is considered a "hit" if its IoU with a ground truth object is at least 0.5.

**Comment by chunduriv** [comment]
@Suzan009,

Can you check below metrics and  let us know if it helps you? 

F2 Score : [tfa.metrics.FBetaScore ](https://www.tensorflow.org/addons/api_docs/python/tfa/metrics/FBetaScore) with`beta=2.0` 

mAP:  [tfr.keras.metrics.MeanAveragePrecisionMetric
](https://www.tensorflow.org/ranking/api_docs/python/tfr/keras/metrics/MeanAveragePrecisionMetric)

Thanks!

**Comment by Suzan009** [comment]
@chunduriv 

actually no. check this,  [tfa.metrics.FBetaScore](https://github.com/tensorflow/addons/blob/v0.15.0/tensorflow_addons/metrics/f_scores.py#L26-L32), it clearly states and it's not suitable for object detection. And same as this one too [tfr.keras.metrics.MeanAveragePrecisionMetric](https://www.tensorflow.org/ranking/api_docs/python/tfr/keras/metrics/MeanAveragePrecisionMetric#compute_output_shape)

Please correct me with a working example if I miss something.

**Comment by sachinprasadhs** [comment]
You can find the details in the supported object detection evaluation protocols [here](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/evaluation_protocols.md).

**Comment by Suzan009** [comment]
Please note that I need keras implementation so that I can integrate it conveniently into my pipeline. That repo is a complete mess for me. I also loss work motivation when I see that repo, no offense.

**Comment by sachinprasadhs** [comment]
@pure-rgb , 
Keras implementation of metric FBetaScore will be available from TensorFlow 2.13 version, [here](tf.keras.metrics.FBetaScore) is the document link for the detail.
For now, you can make use of this using `tf-nightly` version.
Let us know if this addresses your concern. Thanks!
